syntax = "proto3";

package orbax_model_tf_concrete_function_handle;

// A handle (i.e. "pointer") to a TF concrete function stored in a TF
// SavedModel.
message TfConcreteFunctionHandle {
  // The name of the TF concrete function.
  string fn_name = 1;

  // The input/argument names of the TF concrete function. This field is for the
  // case where the stored TF concrete function only accepts keyword-only
  // arguments (e.g. those functions stored in the "serving_signatures" of a TF
  // SavedModel) but we want to allow calling the function with positional
  // arguments.
  //
  // If this field is not set, this function can only be called with
  // keyword-only arguments.
  optional StrList input_names = 20;

  // Similar to `input_names`, but for outputs.
  optional StrList output_names = 30;
}

message StrList {
  repeated string elements = 1;
}
